Libplanet/EnumerableExtensions.cs

```diff
@@ -1,13 +1,60 @@
-#if !NETSTANDARD2_1
 using System;
 using System.Collections.Generic;
+#if !NETSTANDARD2_1
 using System.Diagnostics;
+#endif
 using System.Linq;
 
 namespace Libplanet
 {
     internal static class EnumerableExtensions
     {
+        /// <summary>
+        /// Similar to <see
+        /// cref="Enumerable.Max{TSource,TResult}(IEnumerable{TSource}, Func{TSource, TResult})"/>
+        /// method, but it returns the source element with the maximum value of the given
+        /// <paramref name="selector"/>, instead of the maximum value.
+        /// </summary>
+        /// <param name="source">The source enumerable.</param>
+        /// <param name="selector">A function to select the value to compare.</param>
+        /// <typeparam name="TSource">The type of the source elements.</typeparam>
+        /// <typeparam name="TComparable">The type of the value to compare.</typeparam>
+        /// <returns>The source element with the maximum value of the given
+        /// <paramref name="selector"/>.</returns>
+        /// <exception cref="InvalidOperationException">Thrown when the source sequence is empty.
+        /// </exception>
+        public static TSource Greatest<TSource, TComparable>(
+            this IEnumerable<TSource> source,
+            Func<TSource, TComparable> selector
+        )
+            where TComparable : IComparable<TComparable>
+        {
+#nullable disable
+            bool first = true;
+            TSource greatestElement = default;
+            TComparable greatestValue = default;
+            foreach (TSource element in source)
+            {
+                TComparable value = selector(element);
+                if (first || value.CompareTo(greatestValue) > 0)
+                {
+                    greatestValue = value;
+                    greatestElement = element;
+                }
+
+                first = false;
+            }
+
+            if (first)
+            {
+                throw new InvalidOperationException("Sequence contains no elements.");
+            }
+
+            return greatestElement;
```

Comment:
Is it possible to `null` when `if (first || value.CompareTo(greatestValue) > 0)` is all false?

Reply:
The `greatestElement` must not be `null`, except for empty sequence, and that's why `InvalidOperationException` is thrown if `first` is still `true` after the below loop ends.